### PR TITLE
Codebuild web-hook triggers related fixes

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

- Fix the `publish_to_local_registry` job when e2e is triggered via a web-hook event.
- Fix the `deploy` job when release is triggered via a web-hook event.
- Fix the `lint` job

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .codebuild/scripts/lint_pr.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 set -xeo pipefail
 # extract the PR number from the PR link
 PR_NUM=${CODEBUILD_WEBHOOK_TRIGGER##*/}
+PROJECT_USERNAME=aws-amplify
+REPO_NAME=amplify-codegen
```
Comment (AaronZyLee) [inline .codebuild/scripts/lint_pr.sh:5]: These two can be removed as they are already included in the CodeBuild build environment vars. (can be found in CDK code)

**Review by AaronZyLee — CHANGES_REQUESTED** [review]

File: .codebuild/scripts/lint_pr.sh (hunk 1 above)
Comment (phani-srikar) [reply .codebuild/scripts/lint_pr.sh:5]: Was it a recent CDK update? I ran this one today morning and it failed to resolve these variables correctly.

Comment (AaronZyLee) [reply .codebuild/scripts/lint_pr.sh:5]: Looks like these params are defined for PR project but for release. I will take my words back. Either approach works for me.

Comment (phani-srikar) [reply .codebuild/scripts/lint_pr.sh:5]: This fails on the deployment project which doesn't have these as ENV vars.

**Review by sundersc — APPROVED** [review]

**Review by AaronZyLee — APPROVED** [review]

**Review by alharris-at — APPROVED** [review]

**Pull request merged**